ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -0,0 +1,251 @@
+          dbTracker =
+              new LoadDatabase(context, dbEvent, work.dbNameToLoadIn, loadTaskTracker)
+                  .tasks();
+          loadTaskTracker.update(dbTracker);
+          if (work.hasDbState()) {
+            loadTaskTracker.update(updateDatabaseLastReplID(maxTasks, context, scope));
+          }
+          work.updateDbEventState(dbEvent.toState());
+          scope.database = true;
+          scope.rootTasks.addAll(dbTracker.tasks());
+          dbTracker.debugLog("database");
+          break;
+        case Table: {
+          /*
+              Implicit assumption here is that database level is processed first before table level,
+              which will depend on the iterator used since it should provide the higher level directory
+              listing before providing the lower level listing. This is also required such that
+              the dbTracker /  tableTracker are setup correctly always.
+           */
+          TableContext tableContext =
+              new TableContext(dbTracker, work.dbNameToLoadIn, work.tableNameToLoadIn);
+          TableEvent tableEvent = (TableEvent) next;
+          LoadTable loadTable = new LoadTable(tableEvent, context, tableContext);
+          tableTracker = loadTable.tasks();
+          if (!scope.database) {
+            scope.rootTasks.addAll(tableTracker.tasks());
+            scope.table = true;
+          }
+          setUpDependencies(dbTracker, tableTracker);
+          /*
+            for table replication if we reach the max number of tasks then for the next run we will
+            try to reload the same table again, this is mainly for ease of understanding the code
+            as then we can avoid handling == > loading partitions for the table given that
+            the creation of table lead to reaching max tasks vs,  loading next table since current
+            one does not have partitions.
+           */
+
+          // for a table we explicitly try to load partitions as there is no separate partitions events.
+          LoadPartitions loadPartitions =
+              new LoadPartitions(context, tableTracker, tableEvent, work.dbNameToLoadIn,
+                  tableContext);
+          TaskTracker partitionsTracker = loadPartitions.tasks();
+          partitionsPostProcessing(iterator, scope, loadTaskTracker, tableTracker,
+              partitionsTracker);
+          tableTracker.debugLog("table");
+          partitionsTracker.debugLog("partitions for table");
+          break;
+        }
+        case Partition: {
+          /*
+              This will happen only when loading tables and we reach the limit of number of tasks we can create;
+              hence we know here that the table should exist and there should be a lastPartitionName
+          */
+          PartitionEvent event = (PartitionEvent) next;
+          TableContext tableContext = new TableContext(dbTracker, work.dbNameToLoadIn,
+              work.tableNameToLoadIn);
```

Comment:
here it wont matter because the partitions class is taking loadTaskTracker as the limiter, and the tracker in context is used for some other purpose;